test/hotspot/jtreg/compiler/codegen/BMI2.java

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Support BMI2 instructions on x86/x64
+ *
+ * @run main/othervm -Xbatch -XX:-TieredCompilation
+ *      -XX:CompileCommand=compileonly,compiler.codegen.BMI2$BMITests::*
+ *      compiler.codegen.BMI2
+ */
+
+package compiler.codegen;
+
+public class BMI2 {
+    private final static int ITERATIONS = 1000000;
+
+    public static void main(String[] args) {
+        int ix = 0;
+        int i = 0;
+
+        { // match(Set dst (ConvI2L (AndI src1 mask)))
+            long z = BMITests.bzhiI2L(ix, i);
+
+            for (int i2 = 0; i2 < ITERATIONS; i2++) {
```

Comment:
I know that you follow BMI.java code example but we need to update these tests. Since you are writing new test you may do it right. See discussion for https://github.com/openjdk/jdk/pull/2811

And, please, don't use 1000,000 iterations. C2 compilation is triggered with 10,000. So 50,000 would be more than enough.

Reply:
Thanks. I will have an update that lowers iterations. I've read through #2811 and will see what I can apply to BMI2.java tests.